arangod/IResearch/IResearchFeature.cpp

```diff
@@ -632,6 +652,35 @@ void IResearchFeature::Async::start() {
     << "started " << _pool.size() << " ArangoSearch maintenance thread(s)";
 }
 
+void IResearchFeature::Async::stop(Thread* redelegate /*= nullptr*/) {
+  _terminate.store(true); // request stop asynchronous tasks
+  notify(); // notify all threads
+
+  CONDITION_LOCKER(lock, _join);
+
+  // join with all threads in pool
+  for (auto& thread: _pool) {
+    if (thread.hasStarted()) {
+      while(thread.isRunning()) {
+        _join.wait();
+      }
+    }
+
+    // redelegate all thread tasks if requested
+    if (redelegate) {
+      SCOPED_LOCK(redelegate->_mutex);
+
+      for (auto& task: thread._pending) {
+        redelegate->_pending.emplace_back(std::move(task));
```

Comment:
The `emplace_back` may throw an exception, but is called from the destructor...
Should be wrapped in a try..catch block either here or the call in the destructor should be wrapped into try..catch.

Reply:
the call from the destructor will have redelegate == nullptr

Reply:
Ok then